lesson22/index.js

```diff
@@ -0,0 +1,242 @@
+        if (!json) return;
+
+        const data = json.data;
+
+        if (data.length === 0) {
+            parent.appendChild(createErrorMessage("まだデータがありません"));
+            console.log("まだデータがありません");
+            return;
+        }
+        return data;
+
+    } catch(e) {
+        console.error(e);
+        parent.appendChild(createErrorMessage(e));
+    } finally {
+        removeLoading();
+    }
+};
+
+const init = async() => {
+    const data = await getUserData();
+    
+    if (data) {
+        renderTable(data);
+        renderSortButton();
+        setButtonForInitDisplay();
+        addEventListenerForSortButtons(data)
+    }
+};
+
+const createErrorMessage = text => {
+    const errorText = createElementWithClassName("p", "error-message");
+    errorText.textContent = text;
+    return errorText;
+};
+
+const renderTable = data => {
+    const table = createElementWithClassName("table", "table");
+    const titleKeys = Object.keys(tableTitlesData);
+    const titleNames = Object.values(tableTitlesData);
+    const tableHead = createTableHead(titleNames);
+    const tableBody = createTableBody(data, titleKeys);
+
+    parent.appendChild(table).appendChild(tableHead).after(tableBody);
+};
+
+const createTableHead = titlesData => {
+    const tableHead = document.createElement("thead");
+    const tr = createElementWithClassName("tr", "table__row");
+    const fragment = document.createDocumentFragment();
+    
+    for(let i = 0; i < titlesData.length; i++){
+        const th = createElementWithClassName("th", "table__title js-table-title");
+        th.textContent = titlesData[i];
+        fragment.appendChild(th);
+    }
+    tableHead.appendChild(tr).appendChild(fragment);
+
+    return tableHead;
+};
+
+const createTableBody = (data, keys) => {
+    const tableBody = document.createElement("tbody");
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < data.length; i++){
+        const tr = createElementWithClassName("tr", "table__row js-table-row");
+        const td = createTableContents(data[i], keys);
+        fragment.appendChild(tr).appendChild(td);
+    }
+    tableBody.appendChild(fragment);
+
+    return tableBody;
+};
+
+const createTableContents = (data, keys) => {
+    const fragment = document.createDocumentFragment();
+
+    for(let i = 0; i < keys.length; i++){
+        const td = createElementWithClassName("td", "table__contents js-table-contents");
+        td.textContent = data[keys[i]];
+        fragment.appendChild(td);
+    }
+
+    return fragment;
+};
+
+const createSortButtons = () => {
+    const ul = createElementWithClassName("ul", "sort-buttons");
+    ul.id = "js-sort-buttons";
+
+    const buttonConfig = [
+        { src: "../img/icon-both.svg", alt: "idを昇順に並び替える", dataSet: "default"},
+        { src: "../img/icon-asc.svg", alt: "idを降順に並び替える", dataSet: "asc"},
+        { src: "../img/icon-desc.svg", alt: "idを順不同に並び替える", dataSet: "desc"}
```

Comment:
IMO
I was taught by moriken-sensei that fixed values that can never be overturned are managed by objects.
```js
const Sort = {
    Default: "default",
    Asc: "asc",
    Desc: "desc",
  }

```
```suggestion
        { src: "../img/icon-both.svg", alt: "idを昇順に並び替える", dataSet: Sort.Default},
        { src: "../img/icon-asc.svg", alt: "idを降順に並び替える", dataSet: Sort.Asc),
        { src: "../img/icon-desc.svg", alt: "idを順不同に並び替える", dataSet: Sort.Desc}
```

Reply:
Thank you so much for sharing this review from your teacher!

> fixed values that can never be overturned are managed by objects.

I learned about it for the first time. I'm learning a lot!

I fixed it,
am I correct in understanding that this is defined as a 'constant' so the first letter is uppercase?

[24f7cd5](https://github.com/haru-programming/JavaScript-lessons/pull/27/commits/24f7cd5b325685dd27072a3e00d3c700abea2871)